pkg/k8shandler/visualization.go

```diff
@@ -204,7 +209,7 @@ func (clusterRequest *ClusterLoggingRequest) removeKibana() (err error) {
 			return
 		}
 
-		if err = clusterRequest.RemoveConfigMap("sharing-config"); err != nil {
+		if err = clusterRequest.RemoveSharedConfigMap(loggingSharedConfigMapName, loggingSharedConfigNs); err != nil {
```

Comment:
Since we may have a scenario where we are upgrading the operator.  I believe we will either have to move this code or have similar code which ALWAYS tries to remove the configmap that was deployed in earlier versions.  As it stands this will only remove the new configmap created by this PR

Reply:
What about the role and rolebinding to give `system:authenticaed` access to the shared config map? Shouldn't we delete them in the old namespace and create them in the new one for the migration path?

Reply:
> What about the role and rolebinding to give `system:authenticated` access to the shared config map? Shouldn't we delete them in the old namespace and create them in the new one for the migration path?

I'm not sure what the 'old' code did exactly but I thought it was to create the shared config in the logging namespace.  Did it also create roles and binding? Maybe specifically for a console service account?  If so, then yes we should remove them.  This PR should:
* rename the shared config
* create shared config in the shared namespace
* create roles/bindings so it can create the config in the appropriate NS
* cleanup all the bits associated with the 'old' location

Reply:
The old code created the shared config in the logging namespace and in addition a Role and Rolebinding for all authenticated users to access this namespace in companion when the kibana route was created/updated. Since we reconciled this role and rolebinding in addition to the configmap, we need to remove it in the logging space and reconciled it in the config-managed namespace.

Ok hold on, I will amend this PR then.

Reply:
Amended reconcilation for shared config role and rolebinding as well as deleting old ones.